rules/config

```diff
@@ -208,6 +208,14 @@ SONIC_ENABLE_IMAGE_SIGNATURE ?= n
 # The absolute path should be provided.
 SONIC_ENABLE_SECUREBOOT_SIGNATURE ?= n
 
+# following flags are used for image secure upgrade verification:
+# SECURE_UPGRADE_DEV_SIGNING_KEY - path to development signing key, used for image signing during build
+# SECURE_UPGRADE_DEV_SIGNING_CERT - path to development signing certificate, used for image signing during build
+# SECURE_UPGRADE_MODE - enum value for secure upgrade mode, valid options are "dev", "prod" and "no_sign"
+#SECURE_UPGRADE_DEV_SIGNING_KEY = 
+#SECURE_UPGRADE_DEV_SIGNING_CERT =  
+SECURE_UPGRADE_MODE = "no_sign"
```

Comment:
Can we only give two options: sign, no_sign? The dev or prod is depended on what kinds of certificates users used, and we can provide a script to generate all dev certificates.

We may be not necessary to provide some options only for DEV.
```
#SECURE_UPGRADE_DEV_SIGNING_KEY = 
#SECURE_UPGRADE_DEV_SIGNING_CERT =  
```
Suggest change as below:
```
#SECURE_UPGRADE_SIGNING_KEY = 
#SECURE_UPGRADE_SIGNING_CERT =  
```

Reply:
This mechanism is part of the feature architecture, as described in its HLD. Doing as your suggestion will change our architecture and we prefer not to do so in this stage. There is a serious demand from our costumers, to allow production signing with their own signing mechanisms. If someone wants to use the current flags they can take the DEV_SIGNING flags and assign it to a different variable or flag.
For your last question - signed image can be used for both fresh installation (from ONIE) or upgrade (from SONIC). In case of ONIE installation - ONIE master is currently not verifying the image  - it can be implemented in the future if required and approved